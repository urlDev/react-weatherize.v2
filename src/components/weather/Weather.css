.weatherSituation {
    display: flex;
    justify-content: space-between;
    margin: 0 auto;
    font-family: 'Roboto', sans-serif;
}

.cityName {
    font-size: 20px;
    margin-top: 0;
    margin-bottom: 5px;
}

.date {
    margin: 0;
    font-size: 15px;
}

.degree {
    font-size: 50px;
    margin: 0;
    margin-right: 10px;
}

.center {
    margin: auto 0;
}

@media (min-width: 900px) {
    .weatherSituation {
        margin: 0;
        max-width: 60%;
        width: 30%;
    }
    .cityName {
        font-size: 40px;
    }
    .degree {
        font-size: 100px;
    }
    .date {
        font-size: 18px;
    }
}