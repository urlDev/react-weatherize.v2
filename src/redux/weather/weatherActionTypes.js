const weatherActionTypes = {
  FETCH_WEATHER_BEGIN: 'FETCH_WEATHER_BEGIN',
  FETCH_WEATHER_SUCCESS: 'FETCH_WEATHER_SUCCESS',
  FETCH_WEATHER_ERROR: 'FETCH_WEATHER_ERROR',
  SAVE_LOCAL_STORAGE: 'SAVE_LOCAL_STORAGE',
};

export default weatherActionTypes;
