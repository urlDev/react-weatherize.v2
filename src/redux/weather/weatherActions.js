/* eslint-disable indent */
import { toast } from 'react-toastify';
import axios from 'axios';

export const fetchWeatherBegin = () => ({
  type: 'FETCH_WEATHER_BEGIN',
});

export const fetchWeatherSuccess = (weather) => ({
  type: 'FETCH_WEATHER_SUCCESS',
  payload: { weather },
});

export const fetchWeatherError = (error) => ({
  type: 'FETCH_WEATHER_ERROR',
  payload: { error },
});

export const fetchWeather = (input = 'Helsinki') => async (dispatch) => {
  dispatch(fetchWeatherBegin());
  try {
    const response = await axios.get(
      `https://cors-anywhere.herokuapp.com/http://api.openweathermap.org/data/2.5/weather?q=${input}&APPID=${process.env.WEATHER_API}&units=metric`,
    );
    const data = await response.data;
    return data.cod === 200
      ? dispatch(fetchWeatherSuccess(data))
      : [
          toast.error(
            data.message &&
              data.message.slice(0, 1).toUpperCase() + data.message.slice(1),
          ),
          dispatch(fetchWeatherError(data.message)),
        ];
  } catch (error) {
    return dispatch(fetchWeatherError(error));
  }
};
