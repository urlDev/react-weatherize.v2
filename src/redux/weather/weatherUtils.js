/* eslint-disable implicit-arrow-linebreak */
/* eslint-disable no-confusing-arrow */

const prevSearchedCities = (prevSearchedCity, searchedCity) =>
  prevSearchedCity.includes(searchedCity)
    ? prevSearchedCity
    : [...prevSearchedCity, searchedCity];

export default prevSearchedCities;
