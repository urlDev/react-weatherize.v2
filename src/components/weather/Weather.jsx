/* eslint-disable react/jsx-one-expression-per-line */
import React from 'react';
import moment from 'moment';
import { useSelector } from '../../tests/__mocks__/reactReduxHooks';

import './Weather.css';

const Weather = () => {
  const weather = useSelector((state) => state.weather.weather);

  return (
    <div className="weather__situation">
      <div className="center">
        <h1 className="degree">
          {weather.weather.main && parseInt(weather.weather.main.temp, 10)}Â°
        </h1>
      </div>
      <div className="center">
        <h1 className="cityName">{weather.weather.name}</h1>
        <h3 className="date">{moment().format('DD.MM.YYYY')}</h3>
      </div>
      <div className="center">
        {weather.weather && (
          <img
            src={`https://openweathermap.org/img/wn/${weather.weather.weather[0].icon}@2x.png`}
            alt=""
          />
        )}
      </div>
    </div>
  );
};

export default Weather;
