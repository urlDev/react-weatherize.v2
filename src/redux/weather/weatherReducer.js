import weatherActionTypes from './weatherActionTypes';

import prevSearchedCities from './weatherUtils';

const initialState = {
  loading: false,
  weather: [],
  error: null,
  storage: [],
};

const weatherReducer = (state = initialState, action) => {
  switch (action.type) {
    case weatherActionTypes.FETCH_WEATHER_BEGIN:
      return {
        ...state,
        loading: true,
      };
    case weatherActionTypes.FETCH_WEATHER_SUCCESS:
      return {
        ...state,
        loading: false,
        weather: action.payload.weather,
        storage: prevSearchedCities(state.storage, action.payload.weather.name),
      };
    case weatherActionTypes.FETCH_WEATHER_ERROR:
      return {
        ...state,
        loading: false,
        error: action.payload.error,
      };
    default:
      return state;
  }
};

export default weatherReducer;
