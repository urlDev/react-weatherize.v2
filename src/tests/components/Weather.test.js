/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow } from 'enzyme';

import Weather from '../../components/weather/Weather';

import { weather } from '../fixtures/weather';
import { mockStore } from '../__mocks__/store';
import * as ReactReduxHooks from '../__mocks__/reactReduxHooks';

describe('Weather component', () => {
  let wrapper;
  let store;

  beforeEach(() => {
    store = mockStore({
      weather,
    });

    jest
      .spyOn(ReactReduxHooks, 'useSelector')
      .mockImplementation((state) => store.getState());

    wrapper = shallow(<Weather store={store} />);
  });

  test('Should render the component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Should get the weather state', () => {
    expect(store.getState().weather).toBeTruthy();
  });
});
