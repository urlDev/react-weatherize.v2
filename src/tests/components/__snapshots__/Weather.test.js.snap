// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Weather component Should render the component correctly 1`] = `
<div
  className="weather__situation"
>
  <div
    className="center"
  >
    <h1
      className="degree"
    >
      20
      Â°
    </h1>
  </div>
  <div
    className="center"
  >
    <h1
      className="cityName"
    >
      Athens
    </h1>
    <h3
      className="date"
    >
      04.11.2020
    </h3>
  </div>
  <div
    className="center"
  >
    <img
      alt=""
      src="https://openweathermap.org/img/wn/02d@2x.png"
    />
  </div>
</div>
`;
